flink-core/src/main/java/org/apache/flink/api/common/typeutils/base/LocalDateComparator.java

```diff
@@ -0,0 +1,187 @@
+
+	public LocalDateComparator(boolean ascending) {
+		this.ascendingComparison = ascending;
+	}
+
+	@Override
+	public int hash(LocalDate value) {
+		return value.hashCode();
+	}
+
+	@Override
+	public void setReference(LocalDate toCompare) {
+		this.reference = toCompare;
+	}
+
+	@Override
+	public boolean equalToReference(LocalDate candidate) {
+		return candidate.equals(reference);
+	}
+
+	@Override
+	public int compareToReference(TypeComparator<LocalDate> referencedComparator) {
+		int comp = ((LocalDateComparator) referencedComparator).reference.compareTo(reference);
+		return ascendingComparison ? comp : -comp;
+	}
+
+	@Override
+	public int compare(LocalDate first, LocalDate second) {
+		int cmp = first.compareTo(second);
+		return ascendingComparison ? cmp : -cmp;
+	}
+
+	@Override
+	public boolean invertNormalizedKey() {
+		return !ascendingComparison;
+	}
+
+	@Override
+	public boolean supportsSerializationWithKeyNormalization() {
+		return false;
+	}
+
+	@Override
+	public void writeWithKeyNormalization(LocalDate record, DataOutputView target) throws IOException {
+		throw new UnsupportedOperationException();
+	}
+
+	@Override
+	public int extractKeys(Object record, Object[] target, int index) {
+		target[index] = record;
+		return 1;
+	}
+
+	@SuppressWarnings("rawtypes")
+	@Override
+	public TypeComparator[] getFlatComparators() {
+		return comparators;
+	}
+
+	@Override
+	public LocalDate readWithKeyDenormalization(LocalDate reuse, DataInputView source) throws IOException {
+		throw new UnsupportedOperationException();
+	}
+
+	@Override
+	public int compareSerialized(DataInputView firstSource, DataInputView secondSource) throws IOException {
+		return compareSerializedLocalDate(firstSource, secondSource, ascendingComparison);
+	}
+
+	@Override
+	public boolean supportsNormalizedKey() {
+		return true;
+	}
+
+	@Override
+	public int getNormalizeKeyLen() {
+		return 6;
+	}
+
+	@Override
+	public boolean isNormalizedKeyPrefixOnly(int keyBytes) {
+		return keyBytes < getNormalizeKeyLen();
+	}
+
+	@Override
+	public void putNormalizedKey(LocalDate record, MemorySegment target, int offset, int numBytes) {
+		putNormalizedKeyLocalDate(record, target, offset, numBytes);
+	}
+
+	@Override
+	public LocalDateComparator duplicate() {
+		return new LocalDateComparator(ascendingComparison);
+	}
+
+	// --------------------------------------------------------------------------------------------
+	//                           Static Helpers for Date Comparison
+	// --------------------------------------------------------------------------------------------
+
+	public static int compareSerializedLocalDate(DataInputView firstSource, DataInputView secondSource,
+			boolean ascendingComparison) throws IOException {
+		int cmp = firstSource.readInt() - secondSource.readInt();
+		if (cmp == 0) {
+			cmp = firstSource.readByte() - secondSource.readByte();
+			if (cmp == 0) {
+				cmp = firstSource.readByte() - secondSource.readByte();
+			}
+		}
```

Comment:
This logic is confusing for me. Internally, the LocalDate takes up 4 bytes, but we are using 6 bytes for comparison?